pkg/blockchain/eth/client.go

```diff
@@ -357,63 +365,72 @@ func (c *Client) CreateRawTx(ctx context.Context, txParams TransactionParams) (*
+			To:    &txParams.ToAddress,
+			Gas:   0,              // wei <-> gas exchange ratio
+			Value: txParams.Value, // amount of wei sent along with the call
+			Data:  txParams.Payload,
+		})
+
+		baseTx.Gas = gasLimit
 
-	if txParams.BaseFee == nil {
-		// since ETH and Polygon blockchain already supports London fork.
-		// no need set special block.
-		baseFee := eip1559.CalcBaseFee(&params.ChainConfig{LondonBlock: big.NewInt(1)}, latestBlockHeader)
-
-		// add 25% to baseFee. baseFee always small value.
-		// since we use dynamic fee transactions we will get not used gas back.
-		b := math.Round(float64(baseFee.Int64()) * feeIncrement)
-		baseFee = big.NewInt(int64(b))
-		txParams.BaseFee = baseFee
-	}
-
-	if txParams.GasTips == nil {
-		_ctx3, cancel3 := context.WithTimeout(ctx, c.Config.RPCResponseTimeout)
-		defer cancel3()
-		gasTip, err := c.client.SuggestGasTipCap(_ctx3)
-		// since hardhad doesn't support 'eth_maxPriorityFeePerGas' rpc call.
-		// we should hardcode 0 as a mainer tips. More information: https://github.com/NomicFoundation/hardhat/issues/1664#issuecomment-1149006010
-		if err != nil && strings.Contains(err.Error(), "eth_maxPriorityFeePerGas not found") {
-			log.Error(ctx, "failed get suggest gas tip: %s. use 0 instead", "err", err)
-			gasTip = big.NewInt(0)
-		} else if err != nil {
-			return nil, fmt.Errorf("failed get suggest gas tip: %v", err)
+		if err != nil {
+			return nil, fmt.Errorf("failed to estimate gas: %v", err)
 		}
-		txParams.GasTips = gasTip
-	}
 
-	maxGasPricePerFee := big.NewInt(0).Add(txParams.BaseFee, txParams.GasTips)
-	baseTx := &types.DynamicFeeTx{
-		To:        &txParams.ToAddress,
-		Nonce:     *txParams.Nonce,
-		Gas:       gasLimit,
-		Value:     txParams.Value,
-		Data:      txParams.Payload,
-		GasTipCap: txParams.GasTips,
-		GasFeeCap: maxGasPricePerFee,
+		latestBlockHeader, err := c.HeaderByNumber(ctx, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		if txParams.BaseFee == nil {
+			// since ETH and Polygon blockchain already supports London fork.
+			// no need set special block.
+			baseFee := eip1559.CalcBaseFee(&params.ChainConfig{LondonBlock: big.NewInt(1)}, latestBlockHeader)
+
+			// add 25% to baseFee. baseFee always small value.
+			// since we use dynamic fee transactions we will get not used gas back.
+			b := math.Round(float64(baseFee.Int64()) * feeIncrement)
+			baseFee = big.NewInt(int64(b))
+			txParams.BaseFee = baseFee
+		}
+
+		if txParams.GasTips == nil {
+			_ctx3, cancel3 := context.WithTimeout(ctx, c.Config.RPCResponseTimeout)
+			defer cancel3()
+			gasTip, err := c.client.SuggestGasTipCap(_ctx3)
+			// since hardhad doesn't support 'eth_maxPriorityFeePerGas' rpc call.
+			// we should hardcode 0 as a mainer tips. More information: https://github.com/NomicFoundation/hardhat/issues/1664#issuecomment-1149006010
+			if err != nil && strings.Contains(err.Error(), "eth_maxPriorityFeePerGas not found") {
```

Comment:
Maybe we can use something like this:
    
```
if errors.Is(err, erthMexPriorityFeePerGasErr) {
        ...
    }
```
and change the error returned by the client

Reply:
I didn't touch this piece of code in this PR. It comes from previous code.